### DISCUSSION TITLE
Trying to figure out how to debug http response code 502

### DISCUSSION THREAD

**Discussion by virusakos** [sindresorhus/got#2001] [Q&A] [answered]
I am making a POST request using the got library and I get the following error after 1 minute:

HTTPError: Response code 502 (Proxy Error)
    at Request.<anonymous> (node_modules/got/dist/source/as-promise/index.js:117:42)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

Looks like some sort of timeout to me but I am not sure if there is anything that can be done from client side or it's a server issue and can't be controlled. Any ideas?

Node version: 12.22.10
Got version: 11.8.3

**Comment by szmarczak** [comment, marked as answer]
I believe it's rather a server issue. 5xx in general are server errors.

**Reply by jnnkky** [reply]
I'm gonna chime in on this one. We've used got to make http POST requests to our api on a regular basis. Those every so often resulted in 502s as the person above describes. We basically tried everything to fix this but to no avail.

The errors only started disappearing after I switched from got to use good old node-fetch. I'm not saying that the switch was definitely the fix and got was somehow causing those 502s to be thrown...but it sure looks like that's the case. 

Should the 502s return I'll be back to update this comment.

**Reply by szmarczak** [reply]
Got can't cause the server to return 5xx. It's a client so the fault is definitely on the server side.
